src/server/src/main/java/io/cassandrareaper/resources/RepairRunResource.java

```diff
@@ -379,22 +392,33 @@ public Response modifyRunState(
           .entity("invalid \"state\" argument: " + state.get())
           .build();
     }
-    final RepairRun.RunState oldState = repairRun.get().getRunState();
+    final RepairRun.RunState oldState = repairRun.getRunState();
+
+    if (intesityStr.isPresent() && oldState == newState && newState == RepairRun.RunState.PAUSED) {
+      final Response intensityValidationResponse = validateIntensity(intesityStr);
+      if (intensityValidationResponse != null) {
+        return intensityValidationResponse;
+      }
+
+      final double intensity = Double.parseDouble(intesityStr.get());
+      return editRun(repairRun, repairUnit, segmentsRepaired, intensity);
```

Comment:
`intensity` parameter will be ignored in case when old state and new state is not PAUSED.
So for next two requests
`PUT state=RUNNING&intensity=null`
`PUT state=RUNNING&intensity=0.7`
response will be `Not Modified`
Just like intensity parameter doesn't exist.
For this case can be more aggresive logic like `Bad Request`. But I don't understand profit of this logic.

Reply:
> > what's the response to the (restful) user when newState != RepairRun.RunState.PAUSED a change in intensity was specified but not fulfilled but the repair run was otherwise modified?
>
> intensity parameter will be ignored in case when old state and new state is not PAUSED.
> So for next two requests
> `PUT state=RUNNING&intensity=null`
> `PUT state=RUNNING&intensity=0.7`
> response will be `Not Modified`
> Just like intensity parameter doesn't exist.
> For this case can be more aggressive logic like `Bad Request`. But I don't understand profit of this logic.

I don't think we need more return codes or logic, eg `Bad Request`.
It's just to review that we arn't going to be returning any response status codes that confuse users…

So the logic is (in order of precedence): 
 
 # if request does not specify new state, then BAD_REQUEST is returned,
 # if the repair run (or unit) is not found, then NOT_FOUND is returned,
 # if there's any other repair runs for the related repair unit and they are RUNNING, then return CONFLICT,
 # if state is unaltered and PAUSED, intensity specified but invalid then BAD_REQUEST is returned,
 # if state is unaltered and PAUSED, intensity specified and valid then OK is returned,
 # if state is unaltered, all other parameters are ignored, and NOT_MODIFIED is returned,
 # if state is altered, but an invalid transition, all other parameters are ignored, and METHOD_NOT_ALLOWED returned,
 # if state is altered, parameters are honoured (intensity possibly changed), and OK returned.

The issue I see here is that there's no catch of when:
 
 - (in step 5) intensity is specified and valid but matches existing value. NOT_MODIFIED should be returned instead of OK.
 - (after step 5) when a state transition is specified as well as a new intensity but that new intensity is ignored.

What do you think about creating a separate REST endpoint for handling updates to intensity, so to simplify the above logic?
I'm open to ideas…